src/conditions/mod.rs

```diff
@@ -1,77 +1,133 @@
-use serde::{Deserialize, Serialize};
+use vector_config::configurable_component;
 
-use crate::{config::component::ComponentDescription, event::Event};
+use crate::event::Event;
 
 mod check_fields;
 pub(self) mod datadog_search;
 pub(crate) mod is_log;
 pub(crate) mod is_metric;
-pub mod not;
 mod vrl;
 
 pub use self::vrl::VrlConfig;
+use self::{
+    check_fields::{CheckFields, CheckFieldsConfig},
+    datadog_search::{DatadogSearchConfig, DatadogSearchRunner},
+    is_log::{check_is_log, check_is_log_with_context},
+    is_metric::{check_is_metric, check_is_metric_with_context},
+    vrl::Vrl,
+};
 
 #[derive(Debug, Clone)]
 pub enum Condition {
-    Not(not::Not),
-    IsLog(is_log::IsLog),
-    IsMetric(is_metric::IsMetric),
-    Vrl(vrl::Vrl),
-    CheckFields(check_fields::CheckFields),
-    DatadogSearch(datadog_search::DatadogSearchRunner),
-
-    // used for benchmarks
-    AlwaysPass,
-    AlwaysFail,
-}
+    /// Matches an event if it is a log.
+    IsLog,
 
-impl Condition {
-    pub(crate) const fn is_log() -> Self {
-        Self::IsLog(is_log::IsLog {})
-    }
+    /// Matches an event if it is a metric.
+    IsMetric,
 
-    pub(crate) const fn is_metric() -> Self {
-        Self::IsMetric(is_metric::IsMetric {})
-    }
+    /// Matches an event with a [Vector Remap Language](https://vector.dev/docs/reference/vrl) (VRL) [boolean expression](https://vector.dev/docs/reference/vrl#boolean-expressions).
+    Vrl(Vrl),
+
+    /// Matches an event against an arbitrary set of predicate/value combinations.
+    CheckFields(CheckFields),
+
+    /// Matches an event with a [Datadog Search](https://docs.datadoghq.com/logs/explorer/search_syntax/) query.
+    DatadogSearch(DatadogSearchRunner),
+
+    /// Matches any event.
+    AlwaysPass,
+
+    /// Matches no event.
+    AlwaysFail,
```

Comment:
Worth noting that these are intended for internal testing/bench usage?

Reply:
I had the thought that it would be ripe for dropping, yes, but figured I'd leave it as a deprecation for the next release and then we can delete it right after.